src/browser-driver-manager.js

```diff
@@ -1,259 +1,281 @@
+ * @param {object} options - Pass-through to installBrowser
+ * @throws {Error} Browser must be chrome
+ * @throws {Error} Browser platform must be detectable
+ * @throws {Error} Build ID must resolve
+ * @throws {Error} Browser must be uninstallable when one is already installed
+ * @throws {Error} Version of chrome and chromedriver must be findable
+ * @throws {Error} Environment file must be removable when one already exists
+ * @returns
  */
-function runBashScript({ returnValue = false, sudo = false, args } = {}) {
-  return new Promise((resolve, reject) => {
-    const lines = [];
-    let lastChunk;
+async function install(browserId, options) {
+  const [browser, version = 'latest'] = browserId.split('@');
+
+  // Should support for other browsers be added, commander should handle this check.
+  // With only one supported browser, this error message is more meaningful than commander's.
+  if (!browser.includes('chrome')) {
+    throw new Error(
+      `The selected browser, ${browser}, could not be installed. Currently, only "chrome" is supported.`
+    );
+  }
 
-    const scriptPath = path.join(__dirname, 'index.sh');
-    let child;
-    if (sudo) {
-      child = spawn('sudo', [scriptPath, ...args]);
-    } else {
-      child = spawn(scriptPath, args);
-    }
+  const platform = detectBrowserPlatform();
 
-    child.stdout.on('data', chunk => {
-      lines.push(chunk);
+  if (!platform) {
+    throw new Error(
+      `Unable to detect a valid platform for ${process.platform} ${process.arch}. Darwin (MacOS, iOS, etc.), Windows, and Linux platforms are supported.`
+    );
+  }
 
-      if (!returnValue) {
-        process.stdout.write(chunk);
-      }
-    });
+  // Get the version to install of both Chrome and Chromedriver
+  const buildId = await resolveBuildId(Browser.CHROME, platform, version);
 
-    // for some reason the install script will output curl download
-    // percent through stderr
-    child.stderr.on('data', chunk => {
-      // mimic download percent bar by manually clearing lines
-      // @see https://stackoverflow.com/questions/17309749/node-js-console-log-is-it-possible-to-update-a-line-rather-than-create-a-new-l
-      if (chunk.indexOf('\r') !== -1) {
-        process.stdout.clearLine();
-      }
-      if (lastChunk) {
-        process.stdout.cursorTo(lastChunk.length - 1);
+  // Get any currently installed version
+  const currentVersion = await getVersion(true);
+  const cacheDir = getBDMCacheDir();
+
+  if (currentVersion) {
+    if (currentVersion === buildId) {
+      console.log(
+        `Chrome and Chromedriver versions ${currentVersion} are already installed. Skipping installation.`
+      );
+      return;
+    }
+    console.log(
+      `Chrome and Chromedriver versions ${currentVersion} are currently installed. Overwriting.`
+    );
+    // Uninstall existing installs
+    for (const browser of ['chrome', 'chromedriver']) {
```

Comment:
I think we should instal the new version first before uninstalling the old version. Otherwise with uninstalling first you can get into a state where you install a valid version, then try to install an invalid one and the old one gets removed and no new one gets installed. It's a bit confusing when that happens.